internal/controller/workspace/workspace.go

```diff
@@ -365,3 +367,23 @@ func op2cd(o []terraform.Output) managed.ConnectionDetails {
 	}
 	return cd
 }
+
+// generateWorkspaceObservation is used to produce v1alpha1.WorkspaceObservation from
+// workspace_type.Workspace.
+func generateWorkspaceObservation(op []terraform.Output) v1alpha1.WorkspaceObservation {
+	wo := v1alpha1.WorkspaceObservation{
+		Outputs: make(map[string]string),
+	}
+	for _, o := range op {
+		if !o.Sensitive {
+			if o.Type == terraform.OutputTypeString {
+				wo.Outputs[o.Name] = o.StringValue()
+				continue
+			}
+			if j, err := o.JSONValue(); err == nil {
+				wo.Outputs[o.Name] = string(j)
+			}
```

Comment:
I realized that it would be better to expose the outputs with their native types rather than stringifying everything, so I made some changes to make that work.  Object outputs must still be converted to JSON as they will not be stored or displayed by K8S without a formal schema specified in the CRD.